src/mbgl/map/vector_tile_data.cpp

```diff
@@ -144,15 +144,15 @@ void VectorTileData::redoPlacement(const PlacementConfig newConfig, const std::f
     if (newConfig != placedConfig) {
         targetConfig = newConfig;
 
-        if (!workRequest) {
-            // Don't start a new placement request when the current one hasn't completed yet, or when
-            // we are parsing buckets.
-            redoPlacement(callback);
-        }
+        redoPlacement(callback);
     }
 }
 
 void VectorTileData::redoPlacement(const std::function<void()>& callback) {
+    // Don't start a new placement request when the current one hasn't completed yet, or when
+    // we are parsing buckets.
+    if (workRequest) return;
+
     workRequest.reset();
```

Comment:
It's ok to remove the other two because `workRequest =` will destruct/cancel the old one, right?